web/src/components/pages/dashboard/challenge/challenge.tsx

```diff
@@ -108,7 +112,15 @@ export default function ChallengePage() {
       {showOnboardingModal && (
         <OnboardingModal
           onRequestClose={() => {
-            setShowOnboardingModal(false);
+            setOnboardingModal(false);
+            if (window.location.search.includes('achievement=1')) {
+              addAchievement(
+                <div>
+                  <p>+50 points</p>
+                  <p>Bonus! You signed up in time for some extra points.</p>
+                </div>
+              );
+            }
```

Comment:
When the user **signs up within the first three days**, the URL will look like `?first=1&achievement=1`. Then I will show toast based on the URL. And put an item `first: true` into the sessionStorage. On the challenge dashboard, I can get the item to check whether or not show the onboarding modal

Reply:
A user is redirected to the `/profile` page before they get to the challenge dashboard. I think it makes more sense to show this notification on `/profile`.

Reply:
Another situation is how to show first_sign_up_within_3_days toast for the newly enrollment of already registered users. Users will be redirected into the home page, without dashboard showing up. Have not deeply talked about that before.

Reply:
Also fine to show the notification on the homepage.

We can show notifications on any page 👍

Reply:
OK, show the first_sign_up_within_3_days notification on the homepage, got it.